toolkit/tools/scheduler/scheduler.go

```diff
@@ -351,9 +351,21 @@ func buildAllNodes(stopOnFailure, canUseCache bool, packagesToRebuild []*pkgjson
 
 		if !stopBuilding {
 			if res.Err == nil {
```

Comment:
I think it's time to extract this logic to a separate function, so we can return on errors instead of setting `stopBuilding` and then having to check the variable later. The caller of the new function can set `stopBuilding` if `err != nil && stopOnFailure`.

Reply:
Agreed, I'll make a note to look at a refactor here.